### DISCUSSION TITLE
Any equivalent to BeforeAfterTestAttribute in V2?

### DISCUSSION THREAD

**Discussion by MindSwipe** [xunit/xunit#2285] [Question] [answered]
I saw that V3 adds a `BeforeAfterTestAttribute` which calls a method and provides the `MethodInfo` of the test before and after running a test. Is there any way to do something similar in V2?

Specifically, I have a `DbContextFactory` which creates a unique DB context for each of my tests being run, and I'd like to destroy that DB context after the test ran. The problem is now that I don't want to manually do this for every test, but that I need access to the fully qualified name of the test method that just ran, i.e `SomeProject.UnitTest.UserTest.GetById` as I'm using that to cache my DB contexts

**Comment by bradwilson** [comment, marked as answer]
It does exist in v2: https://github.com/xunit/xunit/blob/v2/src/xunit.core/Sdk/BeforeAfterTestAttribute.cs
